hypha/apply/tasks/models.py

```diff
@@ -0,0 +1,90 @@
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.core.validators import ValidationError
+from django.db import models
+
+from hypha.apply.users.models import Group, User
+
+from .options import TASKS_CODE_CHOICES
+
+
+class Task(models.Model):
+    code = models.CharField(choices=TASKS_CODE_CHOICES, max_length=50)
+    user = models.ForeignKey(
+        User, blank=True, null=True, on_delete=models.CASCADE, related_name="task"
+    )
+    user_group = models.ManyToManyField(
+        Group,
+        related_name="task",
+    )
+    created_at = models.DateTimeField(auto_now_add=True)
+    related_content_type = models.ForeignKey(
+        ContentType,
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+        related_name="task_related",
+    )
+    related_object_id = models.PositiveIntegerField(blank=True, null=True)
+    related_object = GenericForeignKey("related_content_type", "related_object_id")
```

Comment:
Thoughts on linking an `activity` object that lead to this task. Can't think of an use-case now, but it will be use to track the source activity that lead to these actionable items.